managed/services/agents/actions.go

```diff
@@ -77,14 +77,25 @@ func (s *ActionsService) StartMySQLExplainAction(
 		if res.PlaceholdersCount != uint32(len(placeholders)) {
 			return status.Error(codes.FailedPrecondition, "placeholders count is not correct")
 		}
-
 		q = res.ExplainFingerprint
+
+		s, err := s.qanClient.SchemaByQueryID(ctx, serviceID, queryID)
+		if err != nil {
+			return err
+		}
+		schema = s.Schema
 	default:
 		err := s.qanClient.QueryExists(ctx, serviceID, query)
 		if err != nil {
 			return err
 		}
 		q = query
+
+		s, err := s.qanClient.SchemaByQuery(ctx, serviceID, query)
+		if err != nil {
+			return err
+		}
+		schema = s.Schema
```

Comment:
since we deprecated `query` parameter I think we don't need to fix it here

Reply:
If we dont want to fix it even for case when you sending just query, not query ID then sure no problem. Removed and only SchemaByQueryID left.